### PR TITLE
Added TriviaPiece.isComment

### LINKED ISSUE #2452: Add `TriviaPiece.isComment`
Seems like https://github.com/apple/sourcekit-lsp/pull/1037#discussion_r1472335375 could be useful in swift-syntax in general.

### PR DESCRIPTION
**Pull request by vinu-vanjari**
Resolves #2452

### CODE REVIEW

File: Sources/SwiftParserDiagnostics/SyntaxExtensions.swift [hunk 1]
```diff
@@ -181,6 +181,15 @@ public extension TriviaPiece {
     default: return false
     }
   }
+
+  var isComment: Bool {
```
Comment (Matejkob) [inline Sources/SwiftParserDiagnostics/SyntaxExtensions.swift:185]: Could we consider adding some documentation to clarify the purpose of this property?

Comment (Matejkob) [inline Sources/SwiftParserDiagnostics/SyntaxExtensions.swift:185]: Have you considered marking this property as `public`?

File: Sources/SwiftParserDiagnostics/SyntaxExtensions.swift
Comment (Matejkob) [inline Sources/SwiftParserDiagnostics/SyntaxExtensions.swift:1]: I'm wondering if the `SwiftParserDiagnostics` target is the ideal place for this property. Shouldn't we aim for global access? Perhaps declaring it in the `SwiftSyntax` target would be more appropriate? What do you think?

**Review by Matejkob — COMMENTED** [review]
Thank you for your PR, and welcome to the project! 🎉 I've reviewed your changes and left a few ideas for discussion. I'm looking forward to your thoughts and contributions. Welcome aboard!

**Comment by vinu-vanjari** [comment]
@Matejkob ,
thank you very much for the warm welcome, patient comments and taking your time to guide me.
your comments are very helpful and enlightening!!

---
This time, I have changed CodeGeneration template file which then generates `TriviaPieces.swift` file in `Swift-Syntax`.
I referred to `isWhiteSpace` and `isSpaceOrTab` implementations in the same generator template file.

Then, I ran code generation command which generated changes in `Sources/SwiftSyntax/generated/TriviaPieces.swift` 

Looking forward to your feedback

**Review by ahoppen — COMMENTED** [review]
Thank you. Looks good to me. Could you add an entry to the [release notes](https://github.com/apple/swift-syntax/blob/main/Release%20Notes/511.md)?

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by vinu-vanjari** [comment]
@ahoppen Anything pending from my side? I can't see the full status of windows tests

**Comment by Matejkob** [comment]
> @ahoppen Anything pending from my side? I can't see the full status of windows tests

No need to be concerned about the Windows tests—they can occasionally encounter internal issues. Everything's fine on your end. Thanks for your contribution!

**Comment by bnbarham** [comment]
@swift-ci please test Windows platform

**Pull request merged**